lib/dmv.rb

```diff
@@ -9,8 +11,35 @@ def add_facility(facility)
   end
 
   def facilities_offering_service(service)
-    @facilities.find do |facility|
+    @facilities.find_all do |facility|
       facility.services.include?(service)
     end
   end
+
+  def parse_facilities(data)
+    data.each do |d|
+      d.transform_keys! do |key|
+        key_str = key.to_s
+
+        case
+        when key_str.include?('dmv_office') || key_str.include?('office_name')
+          :name
+        when key_str.include?('phone')
+          :phone
+        when key_str.include?('address__1') # overwriting 'address__1' early prevents bad mod on next line
+          :address2
+        when key_str.include?('address_')
+          :address1
+        when key_str.include?('zip')
+          :zip
+        else
+          key
+        end
+      end
+
+      d[:address] = [d[:address1], d[:address2], d[:city], d[:state], d[:zip]].join(' ')
+
+      @facilities << Facility.new(d)
+     end
+  end
 end
```

Comment:
Great job in this class, your `parse_facilities` method is superb. Great use of the `case` statement. I'm not a huge fan of single-letter variable names, however 😉.

I will add that you can eliminate the need to shovel newly created `Facility` objects into an array but switching out `.each` for `.map`.